File: 1. 단계별로_풀어보기/10~19/13. 정렬/8. 11651.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())

axisArr = [list(map(int, input().split())) for _ in range(N)]

axisArr.sort(key=lambda x: (x[1], x[0]))

for axis in axisArr:
  output(f"{axis[0]} {axis[1]}\n")

File: 1. 단계별로_풀어보기/10~19/13. 정렬/8. 11651.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const N = +inputArr[0]
const axisArr = inputArr.slice(1).map(inputStr => inputStr.split(' ').map(Number))

// 정렬
axisArr.sort((axis0, axis1) => {
  if (axis0[1] < axis1[1]) return -1
  else if (axis0[1] === axis1[1] && axis0[0] < axis1[0]) return -1
  else return 1
})

// 출력
axisArr.forEach(axis => process.stdout.write(`${axis[0]} ${axis[1]}\n`))
